test/commands/deploy.test.ts

```diff
@@ -168,4 +344,103 @@ describe('remote deploy environment', function () {
     .it('Creates a deployment when env exists with only env flag', ctx => {
       expect(ctx.stdout).to.contain('Deployed')
     })
+
+
+  const env_config_with_interfaces = {
+    components: {
+      'examples/hello-world': 'latest',
+      'examples/hello-other-world': 'latest'
+    },
+    interfaces: {
+      echo: "${{components['examples/hello-world'].interfaces.echo.url}}",
+      'echo-dos': "${{components['examples/hello-world'].interfaces.echo-dos.url}}",
+      'echo-other': "${{components['examples/hello-other-world'].interfaces.echo-other.url}}"
+    },
+  };
+
+  const hello_world_component_digest = {
+    "id": "37a01809-a2e8-4000-b211-fa39d1d21ffc",
+    "tag": "latest",
+    "config": {
+      "name": "examples/hello-world",
+      "services": {
+        "api": {
+          "interfaces": {
+            "main": 3000
+          },
+        },
+        "api-dos": {
+          "interfaces": {
+            "main": 3000
+          },
+        }
+      },
+      "interfaces": {
+        "echo": {
+          "url": "${{ services.api.interfaces.main.url }}"
+        },
+        "echo-dos": {
+          "url": "${{ services.api-dos.interfaces.main.url }}"
+        },
+        "echo-other": {
+          "url": "${{ dependencies['examples/hello-other-world'].interfaces.echo-other.url }}"
+        }
+      },
+      "dependencies": {
+        "examples/hello-other-world": "latest"
+      }
+    },
+    "component": {
+      "id": "2638e8a4-0cdf-4581-b59a-2120d3616e03",
+      "name": "hello-world",
+    }
+  };
+
+  const hello_world_edges = [
+    {
+      "to": "examples/hello-other-world/api:latest",
+      "interfaces_map": {
+        "echo-other": "main"
+      }
+    },
+    {
+      "to": "examples/hello-world/api:latest",
+      "interfaces_map": {
+        "echo": "main"
+      }
+    },
+    {
+      "to": "examples/hello-world/api-dos:latest",
+      "interfaces_map": {
+        "echo-dos": "main"
+      }
+    }
+  ];
+
+  let deploy_spy: SinonSpy;
+  mockArchitectAuth.stub(DeployCommand, 'POLL_INTERVAL', () => { return 0 })
+    .do(ctx => {
+      deploy_spy = sinon.fake.returns(undefined);
+      sinon.replace(DeployCommand.prototype, 'deployRemote', deploy_spy);
```

Comment:
Don't use sinon in the do since the restore might not fire if an expect fails. 
You can use stub without defining outside of the test scope:
```ts
mockArchitectAuth
    .stub(docker, 'buildImage', sinon.stub().returns('repostory/account/some-image:1.0.0'))
    ...
    .it('test file: replacement', ctx => {
      const buildImage = docker.buildImage as sinon.SinonStub;
      expect(buildImage.notCalled).to.be.true;
    });
```